### PR TITLE
feat(tencent): Support UserData for tencent provider

### PR DESCRIPTION
**Pull request by zsnmwy**
> 在创建云服务器时，您可以通过指定自定义数据，进行配置实例。当云服务器首次启动时，自定义数据将以文本的方式传递到云服务器中，并执行该文本。如果您一次购买多台云服务器，自定义数据会在所有的云服务器首次启动时运行该文本。
> https://cloud.tencent.com/document/product/213/17525
> https://cloud.tencent.com/document/product/213/15730

Easy for us to initialize the system.

### CODE REVIEW

**Review by Jason-ZW — CHANGES_REQUESTED** [review]
Your commits seem to be problematic. Please remove the merge commit

**Review by Jason-ZW — CHANGES_REQUESTED** [review]
Commits should not contain 'feat(Tencent): Support UserData for Tencent Provider' because it is already in the `master` branch.

**Comment by zsnmwy** [comment]
> Commits should not contain 'feat(Tencent): Support UserData for Tencent Provider' because it is already in the `master` branch.

The `master` branch contains "feat(tencent): Support spot instance for tencent provider https://github.com/cnrancher/autok3s/commit/c2a0e6df19723cbb123177972c9862e1e7708490".

But it doesn't contain 'feat(Tencent): Support UserData for Tencent Provider'.

I fixed the git log. 

PTAL

File: pkg/providers/tencent/tencent.go [hunk 1]
```diff
@@ -601,6 +602,12 @@ func (p *Tencent) deleteInstance(f bool) (string, error) {
 
 // CreateCheck check create command and flags.
 func (p *Tencent) CreateCheck() error {
+	if p.UserDataPath != "" {
+		_, err := os.Stat(p.UserDataPath)
+		if os.IsNotExist(err) {
+			return fmt.Errorf("[%s] calling preflight error: failed to read user data file state %s", p.GetProviderName(), p.UserDataPath)
```
Comment (JacieChao) [inline pkg/providers/tencent/tencent.go:608]: It's better to show file is not exist error. 
```
fmt.Errorf("[%s] calling preflight error: failed to read user data file %s with error %v", p.GetProviderName(), p.UserDataPath, err)
```

**Review by JacieChao — CHANGES_REQUESTED** [review]
Others LGTM

File: pkg/providers/tencent/tencent.go (hunk 1 above)
Comment (zsnmwy) [reply pkg/providers/tencent/tencent.go]: How's that?

```log
FATA[0000] [tencent] calling preflight error: file test.sh is not exist, msg: stat test.sh: no such file or directory 
```

**Review by JacieChao — APPROVED** [review]
@zsnmwy  Thanks for PR

**Pull request merged**